app/uk/gov/hmrc/gform/formtemplate/FormTemplatesControllerRequestHandler.scala

```diff
@@ -164,6 +164,48 @@ object FormTemplatesControllerRequestHandler {
         Json.obj()
       )
 
+    def transformProgressIndicatiorInArray =
+      __.read[JsArray].map { pages =>
+        JsArray(
+          pages.value.map {
+            case page: JsObject if page.value.contains("progressIndicator") =>
+              transformProgressIndicator(page)
+            case x => x
+          }
+        )
+      }
+
+    def transformProgressIndicator(data: JsObject) = {
```

Comment:
Would it be possible to write this in transformers style, i.e. that it returns `Reads[JsObject]` and you know that this act on section json object?